File: core-extra/arazzo-parser/src/main/java/org/evomaster/arazzo/models/domain/FailureReusable.java
package org.evomaster.arazzo.models.domain;

public abstract class FailureReusable {

    public FailureReusable() {
    }

    public static class Failure extends FailureReusable {
        private final FailureAction action;

        public Failure(FailureAction action) {
            this.action = action;
        }

        public FailureAction getAction() {
            return action;
        }
    }

    public static class ReusableObj extends FailureReusable {
        private final Reusable reusable;

        public ReusableObj(Reusable reusable) {
            this.reusable = reusable;
        }

        public Reusable getReusable() {
            return reusable;
        }
    }
}

File: core/src/main/kotlin/org/evomaster/core/problem/rest/arazzo/models/FailureReusable.kt
package org.evomaster.core.problem.rest.arazzo.models

sealed class FailureReusable {
    data class Failure(val action: FailureAction) : FailureReusable()

    data class ReusableObj(val reusable: Reusable) : FailureReusable()
}